### ISSUE TITLE
Should `Move Line Up/Down` move whole folded segments?

### ISSUE THREAD

**Issue by petrbouchal** [notaapp/talk#343]
When I fold a section by clicking e.g. the `h2`, then place the cursor in the line and Move Line Up / Move Line Down (via custom keyboard shortcut), the section unfolds and only the heading line moves up/down.

I think that (and would find it more convenient if) the whole folded segment was moved by the Move Line commands.

**Comment by astoilkov** [comment]
Yeah, absolutely. We've done this for lists but not for headings. It's a little harder for them. Do you know of another editor that does this well? Thanks.

**Comment by petrbouchal** [comment]
I know RStudio does - it's based on ACE I think?

I thought VSCode did but now I found it also cannot move folded blocks in Markdown - they get unfolded once the line is moved.

**Comment by mdefazio** [comment]
Adding a +1 for this.  Noticed Obsidian doesn't do this either, but this would be a very nice feature

**Comment by astoilkov** [comment]
@mdefazio Yeah, I agree. We should do it!
